README.md

```diff
@@ -569,6 +569,8 @@ Note: This example is equivalent to not specifying a mapping or specifying an em
 
 When a `ClusterWorkloadResourceMapping` is defined in the cluster matching a workload resource it **MUST** be used to map the binding that type. If no mapping is available for the type, the implementation **MUST** treat the workload resource as a `PodSpec`-able type.
 
+If a `ClusterWorkloadResourceMapping` resource is modified, and workload resources have had binding data injected using that mapping, then any modifications to these workload resources **MUST** use the new `ClusterWorkloadResourceMapping` definitions.  The existing mapping **MUST** be used to unbind existing resources, and the new mapping **MUST** be used to reproject these existing bindings.
```

Comment:
Wordsmithing the proposal. Incorporate as much (or little) as you agree with.

```suggestion
After a `ClusterWorkloadResourceMapping` resource is modified and workload resources have bindings projected using that mapping, the mapping used to project the binding **MUST** be used to unproject that binding, and the new mapping used to reproject the binding.
```

Reply:
I am finding the words `reproject` and `unproject` a bit confusing. Can we use more descriptive words?
For example, `reproject = project again` & `unproject = remove projection`.

Reply:
Another attempt at wordsmithing:
```suggestion
After a `ClusterWorkloadResourceMapping` resource is modified, existing bindings using the associated workload resource type **MUST** have their projections removed using the old mapping, and each binding **MUST** be reattempted using the new mapping.
```

Reply:
I'm not a fan of "old" and "new" as it's possible to have multiple quick updates for a mapping. We want to remove a binding projection using the same mapping config that was use to create the projection, and then use the latest mapping to create the projection again. Depending on what actually changed in the mapping, this may be a no-op.

Taking another crack, I split the requirement into two independent requirements:
1. binding projections are removed with the same mapping used to originally create the mapping (also meaningful for deleting a ServiceBinding)
2. ClusterWorkloadResourceMapping logically remove and the reapply the binding projection (these are logical operations that can occur in a single k8s api patch/update)

```suggestion
When no longer needed, binding projections **MUST** be removed from a workload using the same mapping originally used to project binding. After a `ClusterWorkloadResourceMapping` resource is modified, each binding targeting the mapped workload type **MUST** be removed, and then reapplied with the latest mapping.
```

Reply:
To throw another thing in the mix (sorry! 😄 ) , I think that `service binding resource is deleted` is limiting to just the case where the `ServiceBinding` CR is removed - but the removal of projection can also occur if the Provisioned Service is removed.  So maybe:

```
When a service binding projection is removed the controller **MUST** use the same mappings from the projection creation.  After a `ClusterWorkloadResourceMapping` resource is modified, each binding targeting the mapped workload type **MUST** be removed, then reattempted with the latest mapping.  If the workload mapping was deleted, then binding **MUST NOT** be reattempted.
```

Reply:
> I think we should force unbinding on removal: having behavior determined by a resource that doesn't exist feels like surprising behavior, as future attempts at binding would fail, even though existing resources have successful bindings. 

I might be overfitting, but I've been trying to avoid overt language about when something should happen in order to be conformant with the spec. While most implementations will be k8s controllers that have a stream of events that will process a change within milliseconds, we also want to be open to other lifecycles, like a CLI that add/removes binding projections on a workload. A CLI inherently can't act until it is invoked to perform some action.

> Attempting to rebind with defaults, while also a possibility, most likely will not result in successful bindings, so we should probably disallow that behavior.

Let it fail. Report the errors. Someone can choose to fix the mapping, or not.

If we didn't allow this behavior, someone could create an unnecessary mapping that was bad, and then removed it expecting the system to heal itself (as k8s does). The system would refuse to undo the bad binding projections.